src/air/constraints/evaluator.rs

```diff
@@ -83,11 +87,37 @@ impl<'poly, F: IsFFTField, A: AIR + AIR<Field = F>> ConstraintEvaluator<'poly, F
         #[cfg(debug_assertions)]
         let mut transition_evaluations = Vec::new();
 
+        let mut boundary_polys_evaluations = Vec::with_capacity(boundary_polys.len());
+        for poly in boundary_polys.iter() {
+            let evaluations = evaluate_polynomial_on_lde_domain(
+                poly,
+                domain.blowup_factor,
+                domain.interpolation_domain_size,
+                &domain.coset_offset,
+            )
+            .unwrap();
+            boundary_polys_evaluations.push(evaluations);
+        }
+
+        let mut boundary_zerofiers_inverse_evaluations =
+            Vec::with_capacity(boundary_zerofiers.len());
+        for poly in boundary_zerofiers.iter() {
+            let mut evaluations = evaluate_polynomial_on_lde_domain(
+                poly,
+                domain.blowup_factor,
+                domain.interpolation_domain_size,
+                &domain.coset_offset,
+            )
+            .unwrap();
+            FieldElement::inplace_batch_inverse(&mut evaluations);
+            boundary_zerofiers_inverse_evaluations.push(evaluations);
+        }
```

Comment:
We can evaluate the polynomials before collecting the original vectors, so we halve the memory taken by the whole operation.

Reply:
Good idea. Done!